bouncer/authentication/models.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils.crypto import get_random_string
+
+
+class AuthToken(models.Model):
+    key = models.CharField(_("Key"), max_length=32, primary_key=True, editable=False)
+    user = models.ManyToManyField(
+        settings.AUTH_USER_MODEL, related_name="auth_token", verbose_name=_("User")
+    )
+    created_at = models.DateTimeField(_("Created"), auto_now_add=True)
+
+    class Meta:
+        verbose_name = _("AuthToken")
+        verbose_name_plural = _("AuthTokens")
+
+    def save(self, *args, **kwargs):
```

Comment:
Instead of overriding `save`, please add the `generate_key` result as a default value. Since the `key` is not editable, the default value will be the one used.

See https://docs.djangoproject.com/en/2.1/ref/models/fields/#default